tools/docgen/docgen/fabric_helpers.py

```diff
@@ -0,0 +1,166 @@
+
+import difflib
+import markdown
+from bs4 import BeautifulSoup
+from nbconvert.preprocessors import Preprocessor
+import os
+from urllib.parse import urlparse
+import requests
+import shutil
+
+class LearnDocPreprocessor(Preprocessor):
+    def __init__(self, remove_tags=None, **kwargs):
```

Comment:
nit: perhaps call this tags_to_remove so it doesent get confused as a boolean variable